### PR TITLE
created delete method

### LINKED ISSUE #4: Needs DELETE method
Delete single joke by id should be sufficient.

### PR DESCRIPTION
**Pull request by NoodleSamaChan**
hello!

fixes #4 

please don't hesitate to let me know if there are any issues.
Cheers!

### CODE REVIEW

**Comment by BartMassey** [comment]
Don't have the earlier PR: guessing you deleted it?

So force-push your whole branch, so that I can get all the commits there.

File: src/jokebase.rs [hunk 1]
```diff
@@ -65,6 +67,15 @@ impl JokeBase {
         self.write_jokes()?;
         Ok(())
     }
+
+    pub fn delete(&mut self, given_id: String) -> Result<(), JokeBaseError> {
+        if self.jokemap.contains_key(&given_id) == false {
+            return Err(JokeBaseError::JokeDoesNotExists(given_id));
+        } else {
+            self.jokemap.remove(&given_id);
+            Ok(())
```
Comment (irevoire) [inline src/jokebase.rs]:
```suggestion
            self.jokemap.remove(&given_id);
            self.write_jokes()?;
            Ok(())
```

After updating the jokemap, we should update the on-disk version of the database.
This way, if a crash occurs after the deletion, we’ll be able to restart the database without losing any operations.

**Review by irevoire — CHANGES_REQUESTED** [review]
You should also define a new HTTP route on the frontend in this file that calls the method you just defined on the database: https://github.com/pdx-cs-rust-web/knock-knock/blob/main/src/api.rs

**Comment by NoodleSamaChan** [comment]
hello both!

Thanks very much for your feedback, I'm working on including everything!

@BartMassey terribly sorry I'm still quite new to this, this is actually my first PR, I didn't do any other prior to this one : )

@irevoire I've included the HTTP route following the example of utoipa with Axum: https://github.com/juhaku/utoipa/blob/master/examples/todo-axum/src/main.rs
I'm not getting any error messages so hopefully I did it properly.

Cheers!

**Review by irevoire — APPROVED** [review]
noice

**Comment by BartMassey** [comment]
@NoodleSamaChan @irevoire I've pushed more changes to the API source, sadly. If you can rebase onto these I'll go ahead and merge this.

**Comment by NoodleSamaChan** [comment]
@BartMassey okay it should be all good now! Thanks for your patience, don't hesitate to let me know if there are any questions : )

**Comment by BartMassey** [comment]
Still needed some work, but I went ahead and merged anyhow. I'll rebase my fixes onto the result.

Thanks much for the contribution!

**Pull request merged**